perception/object_recognition/detection/lidar_apollo_instance_segmentation/src/detector.cpp

```diff
@@ -82,43 +85,49 @@ LidarApolloInstanceSegmentation::LidarApolloInstanceSegmentation() : nh_(""), pn
 }
 
 bool LidarApolloInstanceSegmentation::transformCloud(
-  const sensor_msgs::PointCloud2 & input,
-  sensor_msgs::PointCloud2& transformed_cloud,
+  const sensor_msgs::msg::PointCloud2 & input,
+  sensor_msgs::msg::PointCloud2& transformed_cloud,
   float z_offset)
 {
+  // TODO(mitsudome-r): remove conversion once pcl_ros transform are available.
+  pcl::PointCloud<pcl::PointXYZI> pcl_input, pcl_transformed_cloud;
+  pcl::fromROSMsg(input, pcl_input);
+
   // transform pointcloud to tagret_frame
   if (target_frame_ != input.header.frame_id) {
     try {
-      geometry_msgs::TransformStamped transform_stamped;
+      geometry_msgs::msg::TransformStamped transform_stamped;
       transform_stamped = tf_buffer_.lookupTransform(target_frame_, input.header.frame_id,
-                                                     input.header.stamp, ros::Duration(0.5));
+                                                     input.header.stamp, std::chrono::milliseconds(500));
       Eigen::Matrix4f affine_matrix =
         tf2::transformToEigen(transform_stamped.transform).matrix().cast<float>();
-      pcl_ros::transformPointCloud(affine_matrix, input, transformed_cloud);
+      pcl::transformPointCloud(pcl_input, pcl_transformed_cloud, affine_matrix);
       transformed_cloud.header.frame_id = target_frame_;
     } catch (tf2::TransformException &ex) {
-      ROS_WARN("%s",ex.what());
+      RCLCPP_WARN(node_->get_logger(), "%s",ex.what());
       return false;
     }
   } else {
-    transformed_cloud = input;
+    pcl_transformed_cloud = pcl_input;
```

Comment:
If the frames are the same, the transformed_cloud header.frame_id does not get initialised?

Reply:
`pcl_input` should have header information that is copied from `input` in pcl::fromROSMsg function, so it should be copied as well.

Reply:
pcl_input is a PCL data type, it has a header but does not contain information about the ros frame I think.

Reply:
Actually, PCL PointCloud has header like in ROS, and I believe it is also converted in the function.
https://pointclouds.org/documentation/singletonpcl_1_1_point_cloud.html.

Reply:
Ahh, found it! Thanks!
https://pointclouds.org/documentation/structpcl_1_1_p_c_l_header.html